File: api/src/models/workout_routine.py
from pydantic import BaseModel


class WorkoutRoutineResponse(BaseModel):
    id: int
    name: str
    description: str | None = None
    created_at: str
    updated_at: str


class CreateWorkoutRoutineRequest(BaseModel):
    name: str
    description: str | None = None


class UpdateWorkoutRoutineRequest(BaseModel):
    name: str | None = None
    description: str | None = None

File: ui/src/types/WorkoutRoutine.ts
export interface WorkoutRoutine {
    id: number;
    name: string;
    description: string | null;
    created_at: string;
    updated_at: string;
}

export interface WorkoutRoutineCreate {
    name: string;
    description?: string | null;
}

export interface WorkoutRoutineUpdate {
    name?: string | null;
    description?: string | null;
}

export interface RoutineExercise {
    id: number;
    name: string;
    description: string | null;
    muscle_group: string;
    equipment: string | null;
    instructions: string | null;
    sets: number;
    reps: number;
    order_index: number;
    created_at: string;
    updated_at: string;
}
